doc/tutorials/04-lattice_boltzmann/04-lattice_boltzmann_part2.ipynb

```diff
@@ -88,88 +88,97 @@
     "system.actors.add(lbf)\n",
-    "system.thermostat.set_lb(LB_fluid=lbf, seed=123, gamma=lb_friction)\n",
-    "\n",
-    "system.part.add(pos=[0, 0, 0])\n",
-    "\n",
-    "\n",
-    "# Equilibrate\n",
-    "print(\"Equilibrating the system.\")\n",
-    "system.integrator.run(1000)\n",
-    "print(\"Equilibration finished.\")\n",
     "\n",
-    "# Setup observable correlator\n",
-    "pos = ParticlePositions(ids=(0,))\n",
-    "c = Correlator(obs1=pos, tau_lin=16, tau_max=1000, delta_N=1,\n",
-    "    corr_operation=\"square_distance_componentwise\", compress1=\"discard1\")\n",
-    "system.auto_update_accumulators.add(c)\n",
     "\n",
-    "print(\"Sampling started.\")\n",
-    "for i in range(loops):\n",
-    "    system.integrator.run(steps)\n",
-    "\n",
-    "    if i % 1e2 == 0:\n",
-    "        print(\"\\rSampling: {:.0f}%\".format(i * 100. / loops), end=\"\")\n",
-    "\n",
-    "print(\"\\rSampling finished.\")\n",
-    "\n",
-    "c.finalize()\n",
-    "corrdata = c.result()\n",
-    "corr = np.zeros((corrdata.shape[0], 2))\n",
-    "corr[:, 0] = corrdata[:, 0]\n",
-    "corr[:, 1] = (corrdata[:, 2] + corrdata[:, 3] + corrdata[:, 4]) / 3\n",
+    "system.part.add(pos=[0, 0, 0])\n",
     "\n",
-    "np.savetxt(\"./msd_{:.1f}.dat\".format(lb_friction), corr)\n"
+    "# Setup observable\n",
+    "pos = espressomd.observables.ParticlePositions(ids=(0,))\n",
+    "\n",
+    "# Run for different friction constants\n",
+    "lb_gammas = [1.0, 2.0, 4.0, 10.0]\n",
+    "msd_results = []\n",
+    "\n",
+    "for gamma in lb_gammas:\n",
+    "    system.auto_update_accumulators.clear()\n",
+    "    system.thermostat.turn_off()\n",
+    "    system.thermostat.set_lb(LB_fluid=lbf, seed=123, gamma=gamma)\n",
+    "    # Equilibrate\n",
+    "    logging.info(\"Equilibrating the system.\")\n",
+    "    system.integrator.run(1000)\n",
+    "    logging.info(\"Equilibration finished.\")\n",
+    "    # Setup observable correlator\n",
+    "    correlator = espressomd.accumulators.Correlator(obs1=pos, tau_lin=16, tau_max=LOOPS * STEPS, delta_N=1,\n",
+    "        corr_operation=\"square_distance_componentwise\", compress1=\"discard1\")\n",
+    "    system.auto_update_accumulators.add(correlator)\n",
+    "\n",
+    "    logging.info(\"Sampling started for gamma = {}.\".format(gamma))\n",
+    "    for i in range(LOOPS):\n",
+    "        system.integrator.run(STEPS)\n",
+    "    correlator.finalize()\n",
+    "    msd_results.append(correlator.result())\n",
+    "\n",
+    "logging.info(\"Sampling finished.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Plotting the results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "plt.rcParams.update({'font.size': 22})\n",
+    "plt.ion() # switch on interactive mode\n",
```

Comment:
It is a known limitation in the Jupyter backend that the `plt.show()` command prints a string representation of the python object rather than creating a canvas when it's in the same cell as `import matplotlib.pyplot`, forcing the user to execute the cell a second time to get the plot displayed. There are two ways to remediate this: either split the cell to have the import statement separated from the plotting command, or inserting a `%matplotlib inline` before the import statement. If you choose the latter, you'll need to add the following snippet before the `return code` statement of `disable_matplotlib_gui()`:
```python
    # remove magic function
    code = code.replace(
        "get_ipython().run_line_magic('matplotlib', 'inline')", "")
```

Reply:
i won't touch this... we have this issue in all tutorials and we should just be aware of the bug.

Reply:
hm, read the doc again... I think we should globally check all tutorials for this